bokehjs/src/lib/models/tools/gestures/box_zoom_tool.ts

```diff
@@ -170,6 +174,13 @@ export class BoxZoomToolView extends GestureToolView {
   }
 
   override _doubletap(_ev: TapEvent): void {
+    if (!this.model.active) {
+      return
+    }
+    if (this.model.zoom_out_gesture != "double_tap") {
+      return
+    }
+
     const {state} = this.plot_view
     if (state.peek()?.type == "box_zoom") {
```

Comment:
Perhaps an orthogonal question, but maybe we should undo any previous `box_zoom`? For example, if `box_zoom` was followed by `pan`, then the current logic wouldn't allow to zoom out. Or maybe we should just zoom out unconditionally?

Reply:
I assume you mean repeatedly undo intervening states all the way back to any most recent box zoom? (e.g. trying to undo just the zoom but not the pan seems confusing at best). That seems reasonable but then this seems to become a more general "undo" action. Just spitballing but something like
```
plot.interactive_undo_gesture = double_tap

# undo everything in between now and the last box zoom event
plot.interactive_undo_events = ["box_zoom"]

# undo everything in between the last box zoom or pan tool event
plot.interactive_undo_events = ["box_zoom", "pan_tool"]
```

Reply:
I agree that if a user has zoomed and then panned, at that point what they would want a double-tap to mean is still zoom out, even if the last interaction was a pan. They are simply trying to step back and see the bigger picture, and I'd guess they'd want that to happen at the current pan location rather than at the original box zoom location.

Reply:
I don't understand what it means to "zoom out" a box zoom tool at a different location, especially if the range is log or categorical. What exactly, is the computation of the new bounds? I don't see how "unzoom" can mean anything other than "go back to the exact recorded bounds before the last box zoom. 

If "zoom out the _current_ viewport" is really the actual ask then I think the solution is actually simpler than this PR entirely: jjust make the existing `ZoomOut` tool be able to be triggered via double tap.

Reply:
How do we establish how much to zoom out? Currently we simply undo the zoom. We could zoom out by a fixed factor, like `ZoomOutTool` does. Alternatively we could find the latest box zoom in plot's state and use that to compute the factor. Perhaps fallback to a fixed factor if no box zoom was performed so far. Though the factor would have to be configurable, as it is the case in `ZoomOutTool`.

Reply:
> we could find the latest box zoom in plot's state and use that to compute the factor.

-1 on this approach

Is the concern with just adding the gesture to existing `ZoomOutTool` that it would require the `ZoomOutTool` to be added separately in order to function?

Reply:
It's certainly valid to try to figure out exactly how much to zoom out, where to zoom out, etc., especially for nonlinear coordinate systems. But I'm just focusing on the experience of a BoxZoom user -- I've zoomed in at a particular location, maybe panned around a bit, and I'm now trying to step back a bit and possibly zoom in somewhere else. Without the double-tap feature, the only options I know are to completely reset the plot (definitely not what I want in most cases) or enable wheel zoom (making this tool very awkward to work with). So as a user I don't actually much care precisely where it zooms out to, just that it does zoom out so that I can zoom in somewhere else. If it doesn't zoom out enough I'd just double-tap again. I think that zooming out to the state prior to the most recent box zoom is probably the most natural, but as long as I _somehow_ zoom out, that's the main thing!

Reply:
So, adding a double-tap gesture to the existing `ZoomOutTool` would be fine?

Reply:
I'm happy for whomever is trying this out to decide that. The "undo" approach sounds like the _most_ correct, but if the other way also feels fine for the user and is easier to do, it sounds fine to me too.

Reply:
I am happy with either of these:

* add double tap option to zoom out tool to perform a fixed zoom out of the current viewport
* add a double tap option to box zoom tool that "undos" all the way to exactly the state before the last box zoom

Those are simple to explain. Trying to compute a dynamic non-constant zoom factor for the current viewport based on the last box zoom, wherever or how long ago that was, is not 